### PR TITLE
Add support for splash for non-grub bootloaders and fix adding crypto_keyfile when it's missing

### LINKED ISSUE #2311: Feature request: module bootloader should check for plymouth and splash (similarly to grubcfg)
Currently when using systemd-boot, bootloader module doesn't check if plymouth is installed and doesn't add "splash" to kernel options.

I suggest replicating the functionality from grubcfg
https://github.com/calamares/calamares/blob/1645874973669475954e039fbfbd5b3d126fa938/src/modules/grubcfg/main.py#L149

### PR DESCRIPTION
**Pull request by eugenesan**
The following enables adding "splash" to cmdline when plymouth is installed (independently of grubcfg) and fixes adding crypto_keyfile when it's missing (for example when luksbootkeyfile is disabled).
Fixes https://github.com/calamares/calamares/issues/2311 and https://github.com/calamares/calamares/issues/2310 respectively.

TLDR;
Improvements for systemd-boot support.
With those fixes, it's possible to configure Kubuntu/Lubuntu installer to utilize 2 partition config (EFI + Encrypted root).

### CODE REVIEW

**Comment by demmm** [comment]
Since it is a common practice to keep commits per module (see https://github.com/calamares/calamares/wiki/Develop-Code, last paragraph), it would be better to split this PR up in one for fstab & one for the bootloader module.

The fstab changes probably won't need much discussing, adding to the bootloader module might.

**Comment by dalto8** [comment]
Yes, the changes for crypto keyfile and the changes for splash are unrelated.  Please split them into 2 PRs.

For the keyfile changes, we should refactor it so we don't set it wrong in the first place, instead of modifying it at the end like that.

**Comment by eugenesan** [comment]
Obsoleted by https://github.com/calamares/calamares/pull/2314 and https://github.com/calamares/calamares/pull/2315

**Comment by eugenesan** [comment]
Obsoleted by https://github.com/calamares/calamares/pull/2314 and https://github.com/calamares/calamares/pull/2315
